### PR TITLE
GraphContextMenu: Fixes add annotation action & data links that cause full page reload 

### LINKED ISSUE #31911: Add Annotation button does not work on old graph panel
**What happened**:
Clicking the annotation button in the old graph does not prompt me to add any data.

https://user-images.githubusercontent.com/29557702/110815973-379c7e80-828b-11eb-9432-4e5bb14f6b2d.mov

**What you expected to happen**:
It should show the add annotation form.
**How to reproduce it (as minimally and precisely as possible)**:
Go to dashboard.
Create a panel.
Click on the graph panel to add an annotation.
Click the "add annotation button"

### PR DESCRIPTION
**Pull request by torkelo**
Feels like there should be a better way to close ContextMenu

### CODE REVIEW

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 1]
```diff
@@ -73,7 +67,14 @@ export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClo
...
                 active={item.active}
-                onClick={onClick}
+                onClick={() => {
+                  if (item.onClick) {
+                    item.onClick();
+                  }
+                  if (onClose) {
+                    onClose();
+                  }
+                }}
```
Comment (thisisobate) [inline packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx:77]: Could this work in this scenario?
```
onClick={item.onClick || onClose}
```

**Review by hugohaggmark — APPROVED** [review]
Looks a lot better 🎉

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx (hunk 1 above)
Comment (torkelo) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: @thisisobate  No, that is not the same as the above code. Read the above. What happens to your suggestion when onClick is set? Will onClose be called?

Comment (thisisobate) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: oh yeah true 😅👍
You're right Torkel

**Pull request merged**